e2e/_suites/fleet/services.go

```diff
@@ -161,11 +162,31 @@ func runElasticAgentCommand(profile string, image string, service string, proces
 // into the installer struct, to be used else where
 // If the environment variable ELASTIC_AGENT_DOWNLOAD_URL exists, then the artifact to be downloaded will
 // be defined by that value
+// Else if the environment variable BEATS_LOCAL_PATH is set, then the artifact
+// to be used will be defined by the local snapshot produced by the local build.
 // Else, if the environment variable BEATS_USE_CI_SNAPSHOTS is set, then the artifact
 // to be downloaded will be defined by the latest snapshot produced by the Beats CI.
 func downloadAgentBinary(artifact string, version string, OS string, arch string, extension string) (string, string, error) {
 	fileName := fmt.Sprintf("%s-%s-%s.%s", artifact, version, arch, extension)
 
+	beatsLocalPath := shell.GetEnv("BEATS_LOCAL_PATH", "")
+	if beatsLocalPath != "" {
+		distributions := path.Join(beatsLocalPath, "x-pack", "elastic-agent", "build", "distributions")
+		log.Debugf("Using local snapshots for the Elastic Agent: %s", distributions)
+
+		if extension == "tar.gz" {
```

Comment:
will we support zip eventually?

Reply:
also if extension is not one of supporting we can return a bit earlier before we even try constructing distributions path

Reply:
Indeed. Let's do the validation of supported extensions in a follow-up PR. Do you feel comfortable doing yourself? If not, let's open an issue to capture the work to do